### PR TITLE
[DCA] Add more expvars for custom metrics, and api request rates.

### PR DESCRIPTION
**Pull request by devonboyer**
### What does this PR do?

Add more expvars for custom metrics, and api request rates.

```
"apiv1": {"Metadata": {"Errors": 0, "RequestsPerSecond": 8}},
"custommetrics": {"External": {"Total": 1, "Valid": 0}},
"datadog-api": {"Errors": 2, "QueriesPerHour": 2},
```

### Motivation

More visibility into potential problems.

### Additional Notes

This adds `github.com/paulbellamy/ratecounter` as a dependency.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -794,6 +800,15 @@
...
 
+[[projects]]
+  branch = "master"
+  name = "golang.org/x/mobile"
+  packages = [
+    "asset",
+    "internal/mobileinit"
+  ]
+  revision = "507816974b79c76a5fe70f9580265cd57dc78bbe"
+
```
Comment (devonboyer) [inline Gopkg.lock]: @DataDog/agent-core 

`inv deps` keeps wanting to add this to the `Gopkg.lock`. Should I just remove it?

File: pkg/util/kubernetes/hpa/datadogexternal.go [hunk 2]
```diff
@@ -9,16 +9,31 @@ package hpa
...
+	"github.com/paulbellamy/ratecounter"
+
 	"gopkg.in/zorkian/go-datadog-api.v2"
 
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+var (
+	datadogStats          = expvar.NewMap("datadog")
```
Comment (CharlyF) [inline pkg/util/kubernetes/hpa/datadogexternal.go]: any chance we can be more specific ? `datadog-backend` or `datadog-external` ? Thinking about when we will have `datadog-agent-calls` or the like.

File: pkg/util/kubernetes/hpa/hpa.go [hunk 3]
```diff
@@ -92,11 +92,11 @@ func (c *HPAWatcherClient) run(res string) (added, modified, deleted []*autoscal
 				continue
 			}
 			if rcvdHPA.Type == watch.Added {
-				log.Debugf("Adding this manifest: %s", currHPA)
+				log.Debugf("Added HPA: %s/%s", currHPA.Namespace, currHPA.Name)
 				added = append(added, currHPA)
 			}
 			if rcvdHPA.Type == watch.Modified {
-				log.Debugf("Modifying this manifest: %s", currHPA)
+				log.Debugf("Updated HPA: %s/%s", currHPA.Namespace, currHPA.Name)
```
Comment (CharlyF) [inline pkg/util/kubernetes/hpa/hpa.go]: Should we have this moved to Tracef ? As we realized that HPAs are modified very often and the update is not necessarily relevant.

File: pkg/clusteragent/custommetrics/store.go [hunk 4]
```diff
@@ -107,7 +121,23 @@ func (c *configMapStore) SetExternalMetricValues(added []ExternalMetricValue) er
 		}
 		c.cm.Data[key] = string(toStore)
 	}
-	return c.updateConfigMap()
+	if err := c.updateConfigMap(); err != nil {
```
Comment (CharlyF) [inline pkg/clusteragent/custommetrics/store.go]: We could have metrics that relate to the rate of errors when accessing resources like CM/Nodes/Edpts ... ?
For later, but just to have your thoughts.

File: pkg/util/kubernetes/hpa/datadogexternal.go [hunk 5]
```diff
@@ -9,16 +9,31 @@ package hpa
...
+	datadogErrors         = &expvar.Int{}
+	datadogQueriesPerHour = &expvar.Int{}
+	datadogQueriesCounter = ratecounter.NewRateCounter(1 * time.Hour)
```
[the remark this answers is not in the document]
Comment (devonboyer) [reply pkg/util/kubernetes/hpa/datadogexternal.go]: The counter is stored in memory so it just gets reset.

File: pkg/util/kubernetes/hpa/datadogexternal.go (hunk 2 above)
Comment (devonboyer) [reply pkg/util/kubernetes/hpa/datadogexternal.go]: I like `datadog-api`?

**Review by CharlyF — APPROVED** [review]

**Pull request merged**